src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -79,9 +86,34 @@ public RuntimeConfig GetConfig()
                 subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
         }
 
+        TrySetupConfigFileWatcher();
         return _runtimeConfig;
     }
 
+    /// <summary>
+    /// Checks if we have already attempted to configure the file watcher, if not
+    /// instantiate the file watcher if we are in the correct scenario. If we
+    /// are not in the correct scenario, do not setup a file watcher but remember
+    /// that we have attempted to do so to avoid repeat checks in future calls.
+    /// Returns true if we instantiate a file watcher.
+    /// </summary>
+    private bool TrySetupConfigFileWatcher()
+    {
+        if (!_hasAttemptedFileWatcherConfiguration)
```

Comment:
we either need a boolean here or we have to do something like I had before where we check if the config file watcher is not null. this let's us avoid instantiating an empty config file watcher in the case where we are not actually doing any file watching (in the case that we are not in a hot reload scenario)

Reply:
or we can repeat the check of which scenario we are in every time this gets called. the bool prevents us from doing that more complicated check over and over.

Reply:
In the case where we are do NOT actually WANT to do any file watching, do we even need to call this function? Are we calling it in the late configured scenario? Looking at the name of the function, feels natural to use it only if you KNOW you loaded from a file. 

Would we ever end up in a case where you either need to check for the boolean OR create an empty object?

Reply:
followup from our sync on teams, because of how we check for the runtimeconfig being null and additionally the usage of tryloadconfig, we really dont need this bool. removed for conciseness.